File: p001-p099/leetcode083.cpp
// <!-- encoding UTF-8 --!>
/*****************************************************************************
*                     ----Stay Hungry Stay Foolish----                       *
*   @author	:   Shen                                                         *
*   @name   :   Leetcode 083 Remove Duplicates from Sorted List              *
*****************************************************************************/

#include <bits/stdc++.h>
using namespace std;
typedef long long int64;
template<class T>inline bool updateMin(T& a, T b){ return a > b ? a = b, 1: 0; }
template<class T>inline bool updateMax(T& a, T b){ return a < b ? a = b, 1: 0; }
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

class Solution {
public:
	ListNode* deleteDuplicates(ListNode* head)
	{
		if !(head && head -> next) return head;
		head -> next = deleteDuplicates(head -> next);
		return head -> val == head -> next -> val ? head -> next : head;
	}
};

void test()
{

}

int main()
{

	return 0;
}

File: Archive#00-001-100/083-Remove Duplicates from Sorted List/leetcode083.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
  return head unless head && head.next
  if head.val != head.next.val then
    head.next = delete_duplicates(head.next)
  else
    head = delete_duplicates(head.next)
  end
  return head
end
